### PR TITLE
feat: usm to fum and reverse functions drafted - Fixes #23

### LINKED ISSUE #23: Fund/Defund using USM
Implement in `Proxy` by calling:
```
function fundWithUSM(...) {
    proxy.burn(...);
    proxy.fund(...);
}

function defundToUSM(...) {
    proxy.defund(...);
    proxy.mint(...);
}
```

Note that we are calling the `Proxy` functions, meaning that limit orders are included.

### PR DESCRIPTION
**Pull request by albertocuestacanada**
Simple conversion functions, not sure how useful they are. I'll code the tests if we want to merge this PR.

### CODE REVIEW

**Comment by alexroan** [comment]
Hmmm, I agree that they're simple, but I'm not certain they fall within our concern.

On the other hand, they're in the Proxy, not USM, so it's still a degree of separation between them. 

My worry is that if we start introducing functions like these we're blurring the line between what we set out to create, and trying to imitate platforms that exist for this exact reason (i.e, uniswap to swap between two assets). Do we want to enable a few functions that could distract from what we are trying to achieve in the first place? and prompt us to maybe blur that line even more in future?

Open question 🤷‍♂️

**Comment by albertocuestacanada** [comment]
I'm not too excited about this either, @jacob-eliosoff?

**Comment by jacob-eliosoff** [comment]
See my comments on the [other thread](https://github.com/alexroan/USM/issues/37#issuecomment-701216192).  I think we can probably live without `buy`-type functions, but if we do find cause to add them I don't foresee any problems.

Note that buying is a bit fundamentally distinct from minting/creating (and similarly selling vs burning/redeeming): the latter changes the total qty outstanding, the former doesn't.  Eg, if USM gets anywhere I expect markets like ETH/USM to appear on Uniswap & other exchanges: that would be real buying & selling.  Quite likely it'll be cheaper in many cases than minting/burning via our smart contract, but it's our contract's operations that keep the peg permissionless.

So in short, I wouldn't push this until we hit a real-world need.

**Comment by albertocuestacanada** [comment]
Agreed. I would even go further with adding features to `Proxy`. I would just add the features that we need to have a certain behavior on the UI.

Anyone can build their own proxy, and if we are successful people will do for their own specific use cases, we don't need to do it for them.
